File: accounts/accounts-api/src/main/java/com/hrrm/famoney/accounts/api/AccountDTO.java
package com.hrrm.famoney.accounts.api;

import com.hrrm.famoney.commons.immutables.ImmutableStyle;
import com.hrrm.famoney.jaxrs.DTO;
import com.hrrm.famoney.jaxrs.IdDTO;
import io.swagger.v3.oas.annotations.media.Schema;
import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;

import java.math.BigDecimal;

@Schema(
        name = "Account",
        allOf = { AccountDataDTO.class }
)
@Value.Immutable
@ImmutableStyle
public interface AccountDTO extends DTO, IdDTO, AccountDataDTO {

    @Schema(required = true)
    @NotNull
    Integer getMovementCount();

    @Schema(required = true)
    @NotNull
    BigDecimal getTotal();

}

File: accounts/accounts-api/src/main/kotlin/com/hrrm/famoney/accounts/api/AccountDTO.kt
package com.hrrm.famoney.accounts.api

import com.hrrm.famoney.jaxrs.DTO
import com.hrrm.famoney.jaxrs.IdDTO
import io.swagger.v3.oas.annotations.media.Schema
import java.math.BigDecimal
import java.time.LocalDate

@Schema(name = "Account", allOf = [AccountDataDTO::class])
data class AccountDTO(
    override val id: Int,
    override val name: String,
    override val openDate: LocalDate,
    override val tags: Set<String>,
    @get:Schema(required = true)
    val movementCount: Int,

    @get:Schema(required = true)
    val total: BigDecimal,
) : DTO, IdDTO, AccountDataDTO
